devimint/src/federation.rs

```diff
@@ -711,6 +885,35 @@ async fn set_config_gen_params(
     Ok(())
 }
 
+async fn cli_set_config_gen_params(
+    endpoint: &str,
+    auth: &ApiAuth,
+    mut server_gen_params: ServerModuleConfigGenParamsRegistry,
+) -> Result<()> {
+    self::config::attach_default_module_init_params(
+        BitcoinRpcConfig::from_env_vars()?,
+        &mut server_gen_params,
+        Network::Regtest,
+        10,
+    );
+    // Since we are not actually calling `fedimintd` binary, parse and handle
+    // `FM_EXTRA_META_DATA` like it would do.
+    let mut extra_meta_data = parse_map(
+        &std::env::var(FM_EXTRA_DKG_META_ENV)
+            .ok()
+            .unwrap_or_default(),
+    )
+    .with_context(|| format!("Failed to parse {FM_EXTRA_DKG_META_ENV}"))
+    .expect("Failed");
+    let mut meta = BTreeMap::from([("federation_name".to_string(), "testfed".to_string())]);
+    meta.append(&mut extra_meta_data);
```

Comment:
Could likely use iterator chaining and collecting instead of mutating an existing map.